### PR TITLE
use player classes in player label logic

### PR DESCRIPTION
**Pull request by jonOsm**
labels at the bottom of the screen now use MyCreditPlayer and MyElixirPlayer classes for updates

### CODE REVIEW

File: TypeScriptFrontend/src/game/scenes/main.ts [hunk 1]
```diff
@@ -821,27 +824,29 @@ export class MainScene extends Scene {
...
-                                {
-                                    fontFamily: "Arial",
-                                    fontSize: "23px",
-                                    color: '#000000', // Black color for the number
-                                }
-                            );
-                            capitalNumber.setOrigin(0, 1);
 
-                            this.capitalTexts.push(capitalText, capitalNumber);
+                            if (player instanceof MyCreditPlayer) {
```
Comment (ianlavine) [inline TypeScriptFrontend/src/game/scenes/main.ts:828]: I was thinking this display logic would also be moved out of main. For example OtherPlayer would have display logic for name and count, and then credit player would override the function and also display capital count. Does that make sens?

**Review by ianlavine — COMMENTED** [review]
Looks good. I think the methods for displaying all the players stats (name, nodeCount, optionally CapitalCount) should also be moved out of main.ts if possible.

File: TypeScriptFrontend/src/game/scenes/main.ts (hunk 1 above)
Comment (jonOsm) [reply TypeScriptFrontend/src/game/scenes/main.ts:828]: makes sense! i'll take a look at that tonight

Comment (jonOsm) [reply TypeScriptFrontend/src/game/scenes/main.ts]: changes made, please review most recent commit!

File: TypeScriptFrontend/src/game/objects/otherPlayer.ts [hunk 2]
```diff
@@ -20,5 +27,58 @@ export class OtherPlayer {
     set color(color: readonly [number, number, number]) {
         this._color = color;
     }
+
+    destroyRenderedText() {
+        this.renderedText.forEach((text) => text.destroy())
+    }
+
+    //proposal: remove 'scene' parameter and require 'scene' in constructor
```
Comment (ianlavine) [inline TypeScriptFrontend/src/game/objects/otherPlayer.ts:35]: Yea this makes. I guess its used in a few different functions. I think theres some other classes that would benefit from this change as well.

**Review by ianlavine — APPROVED** [review]

**Pull request merged**